bin/compile

```diff
@@ -134,6 +134,41 @@ else
   CACHED_PYTHON_STACK=$STACK
 fi
 
+# TODO: Move this into a new package manager handling implementation when adding Poetry support.
+if [[ ! -f requirements.txt && ! -f Pipfile && ! -f setup.py ]]; then
+  puts-warn
+  puts-warn "Error: No supported Python package manager files were found."
+  puts-warn
+  # We intentionally don't mention `setup.py` here since it's being removed soon.
+  puts-warn "A Python app on Heroku must have either a 'requirements.txt' or"
+  puts-warn "'Pipfile' file in the root directory of its source code, so the"
+  puts-warn "buildpack knows which dependencies to install."
+  puts-warn
+  puts-warn "Currently the root directory of your app contains:"
```

Comment:
I think the message would be just as clear without listing the files. And this long error message could get very long in a project with many files. I'd suggest going without the directory listing. (optional)

Reply:
So IMO the directory listing is quite an important part of the error message, since in many of the cases users are adamant (eg in support tickets) that the files are present, but it turns out they were miss-spelt or not in the top level directory.

By listing the files in the error message - it gives the user a greater chance to spot the issue, and if they still don't notice it, it means support can diagnose without needing to ask for a screenshot of file listings or having the user push to a support branch.

Re the listing becoming unwieldy - the error message only lists the top level directory (rather than recursively), and the primary target of these error messages is new projects, which typically won't have hundreds of files in the root directory.

It's worth noting that the directory listing is inspired by what the Node.js buildpack already does:
https://github.com/heroku/heroku-buildpack-nodejs/blob/c9931b23b297c71199e2777144f75d58d6d977a5/bin/detect#L64